test/extended/builds/pipeline.go

```diff
@@ -159,24 +169,139 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
+
+			g.By("Pipeline using jenkins-client-plugin")
+
+			g.By("should build and complete successfully", func() {
+				// instantiate the bc
+				g.By("create the jenkins pipeline strategy build config that leverages openshift client plugin")
+				err := oc.Run("create").Args("-f", clientPluginPipelinePath).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// start the build - we run it twice because our sample pipeline exercises different paths of the client
+				// plugin based on whether certain resources already exist or not
+				for i := 0; i < 2; i++ {
+					g.By(fmt.Sprintf("starting the pipeline build and waiting for it to complete, pass: %d", i))
+					br, err := exutil.StartBuildAndWait(oc, "sample-pipeline-openshift-client-plugin")
+					debugAnyJenkinsFailure(br, oc.Namespace()+"-sample-pipeline-openshift-client-plugin", oc, true)
+					if err != nil || !br.BuildSuccess {
+						exutil.DumpBuilds(oc)
+						exutil.DumpBuildLogs("ruby", oc)
+						exutil.DumpDeploymentLogs("mongodb", 1, oc)
+						exutil.DumpDeploymentLogs("jenkins-second-deployment", 1, oc)
+						exutil.DumpDeploymentLogs("jenkins-second-deployment", 2, oc)
+					}
+					br.AssertSuccess()
+
+					g.By("get build console logs and see if succeeded")
+					_, err = j.GetJobConsoleLogsAndMatchViaBuildResult(br, "Finished: SUCCESS")
+					o.Expect(err).NotTo(o.HaveOccurred())
+				}
+
+				g.By("clean up openshift resources for next potential run")
+				err = oc.Run("delete").Args("bc", "sample-pipeline-openshift-client-plugin").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = oc.Run("delete").Args("dc", "jenkins-second-deployment").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = oc.Run("delete").Args("bc", "ruby").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = oc.Run("delete").Args("is", "ruby").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = oc.Run("delete").Args("is", "ruby-22-centos7").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = oc.Run("delete").Args("all", "-l", "template=mongodb-ephemeral-template").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = oc.Run("delete").Args("template", "mongodb-ephemeral").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = oc.Run("delete").Args("secret", "mongodb").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+			})
+
+			g.By("should handle multi-namespace templates", func() {
+				g.By("create additional projects")
+				namespace := oc.Namespace()
+				namespace2 := oc.Namespace() + "-2"
+				namespace3 := oc.Namespace() + "-3"
+
+				err := oc.Run("new-project").Args(namespace2).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = oc.Run("new-project").Args(namespace3).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				defer oc.Run("delete").Args("project", namespace2).Execute()
+				defer oc.Run("delete").Args("project", namespace3).Execute()
```

Comment:
it's a nit, but setup the defer for deleting each project immediately after creating it (so we don't leave project-2 around if creating project-3 fails)